storage_controller/src/reconciler.rs

```diff
@@ -422,8 +441,30 @@ impl Reconciler {
                         progress.bytes_downloaded,
                         progress.bytes_total
                     );
+
+                    if let Some(max_lag) = self.reconciler_config.get_max_secondary_lag_bytes() {
+                        let lag = match progress.bytes_total.checked_sub(progress.bytes_downloaded)
+                        {
+                            Some(lag) => lag,
+                            None => {
+                                return Err(ReconcileError::Other(anyhow!(
```

Comment:
In this case, wouldn't we rather fail-forward and proceed with the migration cutover?  Otherwise we'd fail the reconciliation, it would get re-run later and perhaps keep hitting the same problem forever.

Reply:
I guess this depends if we're in the context of a fill or migration in general... same concern about the Err() case added below.

Reply:
Well the entire point is to not cut over if the lag is too great.

Indeed it would be retried via the background reconcile loop which isn't great. Perhaps we can add a new reconcile error type and roll back the intent state to the observed state when handling this new error in `process_results`.